### ISSUE TITLE
[Wine-tkg] Nonsensical Witcher 3 Performance

### ISSUE THREAD

**Issue by ThisNekoGuy** [Frogging-Family/wine-tkg-git#815]
I don't understand why, but whenever I play Witcher 3 with (Lutris-supplied) DXVK with Wine-tkg (for the GOG version of the game), performance is always inexplicably bad given my hardware; I'm at my wits-end for finding the cause of overall performance loss and constant FPS fluctuations, even with the FSR option enabled in Lutris:

My specs:
**CPU:** AMD Ryzen 7 3700X (16) @ 4.05GHz
**RAM:** 32 GBs @ 3600 MHz
**GPU:** RTX 2080 Ti
**GPU Driver:** 515.57
**DE:** Plasma 5.25.3

Config:
[wine-tkg.cfg.txt](https://github.com/Frogging-Family/wine-tkg-git/files/9214451/wine-tkg.cfg.txt)
(Happens regardless if I use plain or Wine-Staging)

**Comment by Tk-Glitch** [comment]
Common advices:
- Nuke Nvidia caches;
- Nuke DXVK's cache;
- Drop `-O3` (in favor of `-O2`) and `-D_FORTIFY_SOURCE=2` flags, they are both detrimental to Wine's perf;
- Since you're on zen2, in case you're running 5.18.14 or newer kernel, you might consider disabling the perf-impacting retbleed mitigations - https://www.phoronix.com/review/retbleed-benchmark - Also, don't build your kernel with Clang;
- Consider using Nvidia's vulkan dev driver when using dxvk's master to make use of the new pipeline_library additions, improving pipeline-related stutters quite a bit (not compatible with dxvk-async btw);
- If you're using Wayland, try Xorg and vice-versa;
- Make sure your VRAM isn't full - that can happen with some mods;
- Try forcefully disabling your compositor before running the game (alt + shift + F12);
- Try setting a framerate cap lower than your refresh rate through dxvk.conf (see https://github.com/doitsujin/dxvk/blob/master/dxvk.conf#L27);
- If disabled, try enabling vsync;
- Consider trying a different version of the game. All GoG games are leaking file descriptors in Wine (through the galaxy dll), which can lead to stuttering, low perf and even crashes after some time.

Also, what's bad perf to you in this case? The game is known to stutter quite a bit on Windows already so it's unlikely to get better on Linux.

**Comment by ThisNekoGuy** [comment]
> Also, what's bad perf to you in this case? The game is known to stutter quite a bit on Windows already so it's unlikely to get better on Linux.

Basically, it's at 30-40 FPS the majority of the time with frequent violent fluctuations

**Comment by ThisNekoGuy** [comment]
- Also, not sure how to clear the Nviddia cache; I've never done that before :v
- I am using a Clang kernel, so I guess I can rollback to a GCC one or recompile with GCC
- VRAM wasn't even close to full and CPU usage looked low
- The compositor was disabled with Gamemode... though I guess I can try manually
- Question: could DXVK be added to Wine-tkg builds as an option...?

**Comment by ThisNekoGuy** [comment]
Performance seems to be much better though deleting the DXVK cache, recompiling Wine, and switching to a GCC kernel

**Comment by Tk-Glitch** [comment]
> Also, not sure how to clear the Nviddia cache; I've never done that before :v

In your home dir you might find a Nvidia dir, and another one in your `$HOME/.cache` dir.

> Question: could DXVK be added to Wine-tkg builds as an option...?

It's kinda problematic to do but it can be done.

**Comment by ThisNekoGuy** [comment]
Problematic?

**Comment by Tk-Glitch** [comment]
We currently do not have a way to inject and use native dlls in the build system when it comes to plain wine builds, but a module could be made to handle that.

**Comment by ThisNekoGuy** [comment]
@Tk-Glitch Question: The Wine-tkg builds do support FSR, right?

**Comment by Tk-Glitch** [comment]
Yes, but the community patch has to be enabled by the user. Also, since it depends on the proton fs hack, you need it enabled to get the feature.

**Comment by SveSop** [comment]
> > In your home dir you might find a Nvidia dir, and another one in your `$HOME/.cache` dir.
> 
> I only found one in `$HOME` (`~/.nv`), not `$HOME/.cache` apparently :/
> 

Depending on driver and if you have used "older" drivers before, it will either be `$HOME/.nv/` or `$HOME/nvidia` ... or `$HOME/.cache/nvidia` . Since this seems to be a change in drivers and/or also distro dependant, it is rather confusing and not easy to track. (See also: [nvidia variables](https://download.nvidia.com/XFree86/Linux-x86_64/510.39.01/README/openglenvvariables.html) )
However, having 1 "global" nvidia cache file might be neat and all that, but there is some limitations of size (atleast default) that can cause issues with stuttering as it has to overwrite "old" data. Especially if playing more than 1-2 games, i personally would recommend doing:
`export __GL_SHADER_DISK_CACHE_PATH=/home/username/winecache/gamename` for each game you run so that each game (or prefix or whatever) gets its own separate cache file. That way it is less chance of needing to overwrite due to it getting to large, AND troubleshooting by deleting the cache file is easier.
Couple that with `export DXVK_STATE_CACHE_PATH=/home/username/winecache/gamename` and you have both dxvk cache AND nvidia cache in one location for easy clearing. (Foldernames is up to you ofc).

**Comment by ThisNekoGuy** [comment]
That would actually be a good idea to automate as a script to pre-pend to Steam's launch options :thinking:

**Comment by SveSop** [comment]
> That would actually be a good idea to automate as a script to pre-pend to Steam's launch options 🤔

I do not think this is such an issue when running through steam, as i suspect the cache-handling there is done a bit different (shader pre-caching and all that). I had the understanding that this issue was with running games using lutris/self-compiled standalone wine, and not through steam?

**Comment by ThisNekoGuy** [comment]
> I had the understanding that this issue was with running games using lutris/self-compiled standalone wine, and not through steam?

Ah, you're right; I just forgot that Steam nhandles ut differently
I suppose the better way for me to have worded it was one applicable to Lutris (for larger libraries)
